crypto/arm_arch.h

```diff
@@ -49,6 +49,10 @@
 #   else
 #    error "unsupported ARM architecture"
 #   endif
+#  else
+#   if defined(__ARM_ARCH)
```

Comment:
Good point @davidben about `_M_ARM64EC` calling convention.

This pair of lines should still be `elif` like the `defined(__GNUC__)` - both for consistency and to make it easier to add to in the future.

Reply:
Which condition do you suggest to use there?

Reply:
@tom-cosgrove-arm is some action needed here?

Reply:
I'd prefer the `elif` approach here. Currently, the PR gives us

```
# if !defined(__ARM_ARCH__)
#  if defined(__CC_ARM)
:
#  elif defined(__GNUC__)
:
#  else        /* added */
#   if defined(__ARM_ARCH)     /* added */
#    define __ARM_ARCH__ __ARM_ARCH    /* added */
#   endif      /* added */
#  endif
# endif
```

but it would be better as

```
# if !defined(__ARM_ARCH__)
#  if defined(__CC_ARM)
:
#  elif defined(__GNUC__)
:
#  elif defined(__ARM_ARCH)     /* add this instead */
#    define __ARM_ARCH__ __ARM_ARCH    /* added */
#  endif
# endif
```

Reply:
It has been changed as suggested